pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,523 @@
+        f = 2.3
+
+        circuit_value = circuit(x, Y)
+
+    It has three variational parameters ``x`` (a scalar) and two entries of ``Y``
+    (an array-like) as well as a tunable frequency ``f`` for the ``qml.RX`` rotation.
+    A first reconstruction job could be with respect to the first entry of ``Y``,
+    which enters the circuit with a single integer frequency:
+
+    >>> nums_frequency = {"Y": {0: 1, 1: 6}}
+    >>> rec = qml.fourier.reconstruct(circuit, {"Y": [0]}, nums_frequency)(x, Y)
+    >>> rec
+    {'Y': {0: <function _reconstruct_equ.<locals>._reconstruction at 0x7f949627d790>}}
+    >>> recon_Y0 = rec["Y"][0]
+    >>> np.isclose(recon_Y0(Y[0]), circuit_value)
+    True
+    >>> np.isclose(recon_Y0(Y[0]+1.3), circuit(x, Y+np.eye(2)[0]*1.3)
+    True
+
+    We successfully reconstructed the dependence on ``Y[0]`` , keeping ``x`` and ``Y[1]``
+    at the values initialized above. Note that even though information about ``Y[1]`` is
+    contained in ``nums_frequency`` , ``ids`` determines which reconstructions are performed.
+    We may do the same for ``Y[1]`` , which enters the circuit with maximal frequency
+    :math:`1+5=4` . We will also track the number of executions needed:
+
+    >>> dev._num_executions = 0
+    >>> rec = qml.fourier.reconstruct(circuit, {"Y": [1]}, nums_frequency)(x, Y)
+    >>> dev.num_executions
+    13
+
+    As expected, we required :math:`2R+1=2\cdot 6+1=13` circuit executions. However, not
+    all frequencies below :math:`f_\text{max}=6` are present in the circuit, so that
+    a reconstruction using knowledge of the spectrum will be cheaper:
+
+    >>> dev._num_executions = 0
+    >>> spectra = {"Y": {1: [0., 1., 4., 5., 6.]}}
+    >>> rec = qml.fourier.reconstruct(circuit, {"Y": [1]}, None, spectra)(x, Y)
+    >>> dev.num_executions
+    9
+
+    If we want to reconstruct the dependence of ``circuit`` on ``x`` , we can not use
+    ``nums_frequency`` if ``f`` is not an integer. One could rescale ``x`` to obtain
+    the frequency :math:`1` again, or directly use ``spectra`` . We will combine this
+    with another reconstruction with respect to ``Y[0]`` :
+
+    >>> dev._num_executions = 0
+    >>> spectra = {"x": {0: [0., f]}, "Y": {0: [0., 1.]}}
+    >>> rec = qml.fourier.reconstruct(circuit, None, None, spectra)(x, Y, f=f)
+    >>> dev.num_executions
+    5
+    >>> recon_x = rec["x"][0]
+    >>> np.isclose(recon_x(x+0.5), circuit(x+0.5, Y, f=f)
+    True
+
+    Note that by convention, the parameter index for a scalar variable is ``0`` and
+    that the frequency :math:`0` always is included in the spectra. Furthermore, we
+    here skipped the input ``ids`` so that the reconstruction was performed for all
+    keys in ``spectra`` . While the reconstruction with a single non-zero frequency
+    costs three evaluations of ``circuit`` for each, ``x`` and ``Y[0]`` , carrying them
+    out at the same position allowed us to save one evaluation and reduce the number
+    of calls to :math:`5`.
```

Comment:
I'm not sure I follow this sentence!

Reply:
I can see that :D Improved it (hopefully)